### PR TITLE
Jack

### PR DESCRIPTION
**Pull request by jackhannah95**
Hi guys,

This is the explorer code hopefully now in a publication ready format. The repository has been deleted and re-created recently, so you might need to delete your local copy and clone it again for it to work. You'll need to create a 'data' folder inside the 'Data-Explorer' folder but other than that I think you should be able to run through it without changing anything.

If you could hold off on merging it until you've both had a chance to review it'd be appreciated, just in case there are any mistakes in there.

Let me know if there are any issues.

Cheers,

Jack

### CODE REVIEW

File: Data-Explorer/ui.R [hunk 1]
```diff
@@ -30,573 +30,574 @@ fluidPage(style = "width: 100%; height: 100%; max-width: 1200px;",
...
+                         are inaccurate due to a system issue with their patient 
+                         management system. Please bear this in mind when looking 
+                         at the data. Further detail about this can be found in the ",
+                         tags$a(href="http://www.isdscotland.org/tpp/data-quality",
+                                "data quality"),
+                         "section.")
+                     )
+            )
+                     )
+                       )
```
Comment (Natalc02) [inline Data-Explorer/ui.R]: Is it only the dates in the notes that are changed here Jack, the massive blocks of red and green all seem the exact same to me?

Comment (jackhannah95) [reply Data-Explorer/ui.R]: That's an old commit, think I only ordered some dates in it. The only significant change to the ui has been to add hyperlinks in the introduction tab.

**Comment by Natalc02** [comment]
Tested the explorer and it looks good and does everything you want it to. For my part I only think the 2 filepaths in data prep need altered other than that it is good to go.

File: Data-Explorer/ui.R [hunk 2]
```diff
@@ -18,503 +18,520 @@
...
+                  
+                  p("When using the data explorer please take the", 
+                    "following factors into consideration:"),
+                  tags$ul(
+                    tags$li("There are issues with the quality of the data", 
+                            "presented. Known issues are summarised in the", 
+                            tags$a(
+                              href = paste0("https://www.isdscotland.org/",
+                                            "Health-Topics/Hospital-Care/",
+                                            "Publications/2018-10-30/",
```
Comment (Natalc02) [inline Data-Explorer/ui.R]: We could add a note to the top of this script that the data quality link will need to be updated every quarter?

Comment (jackhannah95) [reply Data-Explorer/ui.R]: Yeah that's a good idea. I'll add in the note and change the hyperlinks so they're up to date for next month's publication. Although worth bearing in mind when you're testing it that they won't be working links until the publication is out.

File: Data-Explorer/data_preparation.R [hunk 3]
```diff
@@ -60,24 +60,21 @@
...
 base_filepath <- paste0("//stats/SecondaryCare/Quarterly Publication/TPP",
                         "/4_Oct18/data/output/")
 
-RDS_filepath <- paste0("//stats/SecondaryCare/Quarterly Publication/TPP",
-                       "/4_Oct18/data_explorer/")
-
-
-# 1.3 - Load functions script
-source("//stats/SecondaryCare/Quarterly Publication/TPP/functions.R")
+rds_filepath <- ("./Data-Explorer/data/")
```
[the remark this answers is not in the document]
Comment (jackhannah95) [reply Data-Explorer/data_preparation.R]: Yeah you need to create that folder yourself on your desktop when you clone the repository. It's not an ideal solution but it's the least painful workaround.

**Comment by jackhannah95** [comment]
@Natalc02 did you delete the 'Data-Explorer' folder in your copy of the code and move all the scripts into the 'data_explorer' one? The original filepaths worked for me and Marios, but your ones only work for me if I get rid of that sub-folder. Having one folder inside another with practically the same name isn't especially helpful admittedly, so the 'Data-Explorer' one probably should be renamed at some point.

**Pull request merged**